detectors/gcp/go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/GoogleCloudPlatform/opentelemetry-operations-go/detectors/gcp
 
-go 1.23.8
+go 1.23.0
+
+toolchain go1.24.2
```

Comment:
Would this mean that at least go1.24.2 is required to build and compile this project? I'm ok with this, but is this intentional ?